src/java.base/share/classes/java/net/URLDecoder.java

```diff
@@ -86,17 +88,17 @@
      */
     private URLDecoder() {}
 
-    // The platform default encoding
+    // The default charset
     static String dfltEncName = URLEncoder.dfltEncName;
```

Comment:
Perhaps add the value of file.encoding to the StaticProperties either as a string or as the Charset.
That would allow a few different lookups of the property to be simplified.